UPGRADE-GUIDE.md

```diff
@@ -241,6 +241,23 @@ def call(env)
 end
 ```
 
+If you want your code to work with Rack 3 without having to manually lowercase
+each header key used, instead of using a plain hash for headers, you can use
+`Rack::Headers` on Rack 3.
+
+```ruby
+  headers = Rack.release >= '3' ? Rack::Headers.new : {}
```

Comment:
Very few features require version sniffing, and this is a common one.

I could think of two options

1. Backport `Rack::Headers` to Rack 2 as a no-op, e.g. `Rack::Headers = Hash`.
2. Feature detect, e.g. `Rack.const_defined?(:Headers)`.

Do you have any opinion about how we can make this easier?

Reply:
`Rack.const_defined?(:Headers)` or `defined?(Rack::Headers)` is fine with me.

Reply:
Pushed a change to use `defined?(Rack::Headers)`